File: I3A/eclipse-workspace-java/Scala/src/Persona.scala
// Abstract Class Persona
abstract class Persona(val nome: String, val cognome: String, private var _age: Int) {
  if (this._age < 0) throw new IllegalArgumentException

  def age = _age
  def age_=(v: Int): Unit = {
    if (v >= 0) _age = v else throw new IllegalArgumentException
  }

  override def toString: String = { nome + cognome + "(" + age + ")" }
}

File: I3A/eclipse-workspace-java/M4 (Esami)/src/mylist/Persona.java
package mylist;

public class Persona implements Comparable<Persona> {

	int age;
	String nome;

	Persona(int age, String nome) {
		this.age = age;
		this.nome = nome;
	}

	@Override
	public int compareTo(Persona o) {
		if (this.age > o.age)
			return 1;
		if (this.age < o.age)
			return -1;
		return 0;
	}
}
